Src/DependencyCollector/Shared/Implementation/SqlClientDiagnosticSourceListener.cs

```diff
@@ -0,0 +1,149 @@
+
+        public SqlClientDiagnosticSourceListener(TelemetryConfiguration configuration)
+        {
+            this.client = new TelemetryClient(configuration);
+            this.client.Context.GetInternalContext().SdkVersion =
+                SdkVersionUtils.GetSdkVersion("rdd" + RddSource.DiagnosticSourceCore + ":");
+
+            this.subscriber = new SqlClientDiagnosticSourceSubscriber(this);
+        }
+
+        void IObserver<KeyValuePair<string, object>>.OnCompleted()
+        {
+        }
+
+        void IObserver<KeyValuePair<string, object>>.OnError(Exception error)
+        {
+        }
+
+        void IObserver<KeyValuePair<string, object>>.OnNext(KeyValuePair<string, object> evnt)
+        {
+            switch (evnt.Key)
+            {
+                case SqlAfterOpenConnection:
+                {
+                    this.OnAfterOpenConnection(
+                        (SqlConnection) connectionFetcher.Fetch(evnt.Value),
+                        (DateTimeOffset) this.startTimeFetcher.Fetch(evnt.Value),
+                        (TimeSpan) this.durationFetcher.Fetch(evnt.Value));
+
+                    break;
+                }
+            }
+        }
+
+        public const string SqlConnectionCommandText = "Open";
+
+        private void OnAfterOpenConnection(SqlConnection connection, DateTimeOffset timestamp, TimeSpan duration)
+        {
+            var telemetry = new DependencyTelemetry();
+
+            // TODO: Activities?
+
+            this.client.Initialize(telemetry); // TODO: Call this?
+
+            // TODO: Unify with ProfilerSqlConnectionProcessing?
+            telemetry.Name = string.Join(" | ", connection.DataSource, connection.Database, SqlConnectionCommandText);
+            telemetry.Type = RemoteDependencyConstants.SQL;
+            telemetry.Target = string.Join(" | ", connection.DataSource, connection.Database);
+            telemetry.Data = SqlConnectionCommandText;
+            telemetry.Timestamp = timestamp;
+            telemetry.Duration = duration;
+
+            this.client.Track(telemetry);
+        }
+
+        public void Dispose()
+        {
+            if (this.subscriber != null)
+            {
+                this.subscriber.Dispose();
+            }
+        }
+
+        private sealed class SqlClientDiagnosticSourceSubscriber : IObserver<DiagnosticListener>, IDisposable
+        {
+            private readonly SqlClientDiagnosticSourceListener httpDiagnosticListener;
+            private readonly IDisposable listenerSubscription;
+
+            private IDisposable eventSubscription;
+
+            internal SqlClientDiagnosticSourceSubscriber(SqlClientDiagnosticSourceListener listener)
+            {
+                this.httpDiagnosticListener = listener;
+
+                try
+                {
+                    this.listenerSubscription = DiagnosticListener.AllListeners.Subscribe(this);
+                }
+                catch (Exception ex)
```

Comment:
Based on my history with "must never throw", the thing eventually throws one way or the other (e.g. new code in DS will be added by someone who does not know it should not throw and SDK dependency on DS is currently ">X" so new one will be picked up automatically) :).
I would err on a side of logging the error and gracefully degrade.